app/models/group.rb

```diff
@@ -1,18 +1,17 @@
 class Group < ApplicationRecord
-    validates_presence_of :name
+    include Slugable
+    
+    validates_uniqueness_of :name_en
+    validates_presence_of :name_en
 
     has_many :subgroups
     has_one :subgroup, ->{ by_default }, class_name: 'Subgroup'
-    
-    before_create :set_slug
 
     translates :name, :description, :subtitle
 
-    def set_slug
-        self.slug = name.downcase.gsub(/[[:space:]]/, '-')
-    end
-
+    # Returns subgroup name if exist or nil
+    #
     def default_subgroup
-        subgroups&.first&.name
+        subgroup&.name_en
```

Comment:
this one is returned in the payload, so it should be locale-specific rather than _en I think?